### ISSUE TITLE
Can't figure out how to selectively set "Enable.Video.HKSV.Record.Only.Medium".

### ISSUE THREAD

**Issue by jfmach** [hjdhjd/homebridge-unifi-protect#928]
**Describe The Problem:**
I am trying to set the recording quality to Medium for my two G4 Pro cameras because they keep timing out on High, but I want to keep it High for the other lower resolution cameras since it works fine.
Problem is it doesn't seem to work the way I do it. The logs still show recording quality set to High.

**Logs:**
Startup shows rcording quality is still set to high for the two G4 Pro cameras despite setting them to medium in the configuration:
```
[24/04/2023, 13:55:55] Loaded plugin: homebridge-unifi-protect@6.4.4
[24/04/2023, 13:55:55] Registering platform 'homebridge-unifi-protect.UniFi Protect'
[24/04/2023, 13:56:05] [homebridge-unifi-protect] Launched child bridge with PID [edited]
[24/04/2023, 13:56:09] Registering platform 'homebridge-unifi-protect.UniFi Protect'
[24/04/2023, 13:56:09] [homebridge-unifi-protect] Loaded homebridge-unifi-protect v6.4.4 child bridge successfully
[24/04/2023, 13:56:14] Homebridge v1.6.0 (HAP v0.11.0) (homebridge-unifi-protect) is running on port [edited].
[24/04/2023, 13:56:16] [homebridge-unifi-protect] 192.168.2.4: Connected to the UniFi Protect API at [edited].
[24/04/2023, 13:56:16] [homebridge-unifi-protect] Video Recorder [UNVR4]: Enabled system information sensor: cpu temperature.
[24/04/2023, 13:56:16] [homebridge-unifi-protect] Video Recorder: Discovered nvr: Video Recorder [UNVR] (address: [edited] mac: E4388355EB5E).
[24/04/2023, 13:56:16] [homebridge-unifi-protect] Video Recorder: Discovered camera: 02. Front Gate South [G4 Pro] (address: [edited] mac: [MAC ADDRESS 2]).
[24/04/2023, 13:56:16] [homebridge-unifi-protect] Video Recorder: Discovered camera: 01. Front Gate North [G4 Pro] (address: [edited] mac: [MAC ADDRESS 1]).
[24/04/2023, 13:56:16] [homebridge-unifi-protect] Video Recorder [UNVR4] 02. Front Gate South [G4 Pro]: HomeKit Secure Video event recording configured to use only: High.
[24/04/2023, 13:56:17] [homebridge-unifi-protect] Video Recorder [UNVR4] 01. Front Gate North [G4 Pro]: HomeKit Secure Video event recording configured to use only: High.
[24/04/2023, 13:56:17] [homebridge-unifi-protect] Video Recorder [UNVR4] 02. Front Gate South [G4 Pro]: HomeKit Secure Video event recording enabled: 3840x2160@24fps (High), 2000 kbps with a 4 second timeshift buffer.
[24/04/2023, 13:56:17] [homebridge-unifi-protect] Video Recorder [UNVR4] 01. Front Gate North [G4 Pro]: HomeKit Secure Video event recording enabled: 3840x2160@18fps (High), 2000 kbps with a 4 second timeshift buffer.
```

**Homebridge Configuration:**

```json
        {
            "name": "Config",
            "port": [edited],
            "auth": "form",
            "theme": "auto",
            "tempUnits": "f",
            "lang": "auto",
            "platform": "config"
        },
        {
            "controllers": [
                {
                    "address": "[edited]",
                    "password": "[edited]",
                    "username": "[edited]"
                }
            ],
            "options": [
                "Enable.Nvr.SystemInfo",
                "Disable.Doorbell.Messages.FromDoorbell.[edited]",
                "Enable.Doorbell.Trigger.[edited]",
                "Enable.Video.HKSV.Record.Only.High.[UNVR MAC ADDRESS]",
                "Enable.Video.HKSV.Record.Only.Medium.[MAC ADDRESS 1]",
                "Enable.Video.HKSV.Record.Only.Medium.[MAC ADDRESS 2]"
            ],
            "_bridge": {
                "username": "[edited]",
                "port": [edited]
            },
            "platform": "UniFi Protect"
        },
```

**Screenshots:**
Note that I set it directly in the configuration but here is a screenshot of the GUI to show the checkbox is also properly checked (that's the same for both cameras):
<img width="580" alt="Screenshot 2023-04-24 at 2 13 19 PM" src="https://user-images.githubusercontent.com/5497534/234081346-8f798851-c2cd-4f42-964c-b6d5f2edf2c9.png">

**Environment:**

* **Homebridge Version:** 1.6.0
* **Node Version:** v18.6.0
* **Homebridge-UniFi-Protect Plugin Version:** 6.4.4
* **UniFi Protect Version:** 2.7.34
* **UniFi Protect Controller Type:** UNVR
* **UniFi Protect Controller Firmware Version:** I can't see a firmware version? If you mean the Unifi OS version it is UNVR v3.0.16
* **Apple Device and iOS / macOS / iPadOS / tvOS Version:** iOS 16.4.1
* **Operating System and OS Version:** Debian 10.11

**Comment by hjdhjd** [comment]
Remove the option for the UNVR - it's redundant. HBUP will always try to use the highest quality stream it can by default. See if that works - I'll take a closer look to see if there's something else going on when I have some time.

PS: the Protect controller firmware version isn't v3.0.16 - that's the UniFi OS version on the UNVR. You can find the Protect version under the settings for the Protect app (click on the Protect app, then click on the gear icon).

Best of luck.

**Comment by jfmach** [comment]
@hjdhjd that worked, thank you!

**Comment by hjdhjd** [comment]
No worries. Remember: the defaults are almost always the right answer and always target quality and performance. Never set an option that isn't actually necessary. :smile: 

In your case, you're running a high end camera (G4 Pro) on an underpowered system that can't keep up processing a 4k stream in realtime, as you discovered. Enjoy!

**Comment by hjdhjd** [comment]
Yep. Can be a bandwidth issue as well, depending on how your environment is setup. You are pushing a lot of pixels on a G4 Pro. A lot. And HKSV is very very time sensitive...it has very low tolerance for delays. What's your hardware?

**Comment by jfmach** [comment]
The 2 G4 Pro are over the air but I have 10x the bitrate needed and am not saturating on any leg of the network.
<2ms latency between homebridge and the cameras.

Homebridge runs on a puny raspberry Pi 4 with 4GB RAM, so... there's that 😄

I'll work on gathering those metrics when an actual recording event happens to figure out what the bottleneck is.

Thanks for your help!

**Comment by jfmach** [comment]
That's not what I bought the Pi for originally :D
I did not expect someone would release this incredible plugin to support Unifi Protect cameras so well in homebridge :)
I'll look into alternatives. I like the idea of storing high quality motion events in the cloud.

**Comment by hjdhjd** [comment]
LOL. For what it's worth...HKSV records things at low bitrates and quality. We always try to use the highest quality stream we can as a basis for the (ultimately) low bitrate version that HKSV wants. You're not missing out that much. That said, it is quite a bit zippier on an M1/2 Mac...not that HBUP is a slouch now. 😄 

If you're going to be using multiple G4s (I have more than a dozen G4 Pros in my setup), you're going to want beefier hardware. If you've just got the one...I suspect you're fine. Cheers.

**Comment by hjdhjd** [comment]
The edge case you called out related to configuring multiple "only" options at multiple scoping levels is resolved in v6.5.0. Thanks again for the report.
